src/prefect/server/api/concurrency_limits.py

```diff
@@ -62,6 +148,34 @@ async def read_concurrency_limit(
     The `active slots` field contains a list of TaskRun IDs currently using a
     concurrency slot for the specified tag.
     """
+    # V2-first read; fall back to V1
+    async with db.session_context() as session:
+        v2_limit = await cl_v2_models.read_concurrency_limit(
+            session=session, concurrency_limit_id=concurrency_limit_id
+        )
+
+        if v2_limit and v2_limit.name.startswith("tag:"):
+            tag = v2_limit.name.removeprefix("tag:")
+            lease_storage = get_concurrency_lease_storage()
+            active_slots = await get_active_slots_from_leases(
+                v2_limit.id, lease_storage
+            )
+            v1 = await models.concurrency_limits.read_concurrency_limit_by_tag(
+                session=session, tag=tag
+            )
+            if not active_slots and v1:
+                active_slots = list(v1.active_slots)
+            result_id = v1.id if v1 else v2_limit.id
```

Comment:
I don't think this is valid. When we migrate, we should make sure the ID is kept the same between v1 and v2, but the v1 limit will be deleted. We shouldn't be reading a v1 limit here.